lib/path.js

```diff
@@ -37,6 +37,29 @@ function normalizeArray(parts, allowAboveRoot) {
   return res;
 }
 
+// Returns an array with empty elements removed from either end of the input
+// array or the original array if no elements need to be removed
```

Comment:
It's just a little helper function. There's no point in making a copy of an array when the original is all you need.